tests/test_sticky_keys.py

```diff
@@ -397,8 +397,8 @@ def test_sticky_key_deferred(self):
                 {KC.N0, KC.N1, KC.N2},
                 {KC.N0, KC.N1, KC.N2, KC.N3},
                 {KC.N0, KC.N2, KC.N3},
-                {KC.N2, KC.N3},
-                {KC.N3},
+                {KC.N0, KC.N3},
+                {KC.N0},
```

Comment:
You have to explain to me why this is incorrect.
0 tapped -> sticky
1 pressed -> maybe stacked sticky?
2/3 pressed -> 1 is pressed+interrupted -> not sticky, it's held+stacked key
1 released -> release 1 and all stacked stickys in reverse order
(etc)

Reply:
When 2/3 are pressed:
- 0 remains sticky until an interrupting key is released;
- 1 is used in a non-sticky manner(next key is pressed before 1 is released).
When 1 released it's treated non-stickily.
0 only unsticks itself after the last key is released.

Reply:
In other words, 0 and 1 are not a stack, because 1 is held.

Reply:
The stacking behavior is explicitly designed for and written into code, we even explicitly test it. It doesn't need a seperate named state -- if it's intentional then it's not an emergent property.
Semantics aside, the first argument makes sense. Can't remember why held keys also "stacked", doesn't matter. Your suggestion reduces code size and complexity, fixes an issue, and doesn't seem to effectively break anything.

Reply:
The way I conceptualize that -- that was a design criterion, yes, but the code is more about what makes SKs deactivate. As per design criteria SK should not deactivate SK, so the only stacking related thing in the code is aligning timeouts and they can stack without that.

There's also released->sticky part of the `process_key()` method, but it works the same for stacked and non-stacked SKs.